src/test/groovy/com/palantir/gradle/externalpublish/ExternalPublishRootPluginIntegrationSpec.groovy

```diff
@@ -120,13 +121,18 @@ class ExternalPublishRootPluginIntegrationSpec extends IntegrationSpec {
 
         if (type == 'gradle-plugin') {
             subprojectBuildGradle << '''
+                apply plugin: 'java-gradle-plugin'
+                apply plugin: 'groovy'
+                apply plugin: 'com.palantir.external-publish-jar'
```

Comment:
shouldn't this be applied by default by `com.palantir.external-publish-gradle-plugin` ? The `com.palantir.external-publish-jar` sets up the `MavenPublication` see: https://github.com/palantir/gradle-external-publish-plugin/blob/f66f7f25fa357e921d019c325632b82cf66df9c6/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishJarPlugin.java#L33

If I remove this line, then some of the tests will fail with: 
```
* Exception is:
org.gradle.internal.execution.WorkValidationException: A problem was found with the configuration of task ':gradle-plugin:publishPluginMavenPublicationToSonatypeRepository' (type 'PublishToMavenRepository').
  - In plugin 'org.gradle.maven-publish' type 'org.gradle.api.publish.maven.tasks.PublishToMavenRepository' property 'credentials.username' doesn't have a configured value.
    
    Reason: This property isn't marked as optional and no value has been configured.
    
    Possible solutions:
      1. Assign a value to 'credentials.username'.
      2. Mark property 'credentials.username' as optional.```

Reply:
I think the gradle plugin publish plugin also sets up a publication to publish to the Gradle plugins portal? At least they end up in different places. I wonder if this is bit is now wrong/has changed:

https://github.com/palantir/gradle-external-publish-plugin/blob/f125d707df99d5ec06fd28b17fe3e0b18a492921/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishGradlePluginPlugin.java#L51-L57

Honestly I thought it looked weird the other day as `ExtraPropertiesExtension` is super old/deprecated.

Reply:
Apparently this is still the way? https://docs.gradle.org/current/userguide/publishing_gradle_plugins.html

Reply:
I think the credentials setup  is done per repo level (sonatypeRepo) and is configured by the nexus-publish plugin: https://github.com/palantir/gradle-external-publish-plugin/blob/develop/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishRootPlugin.java#L48-L57. This means that `publish<publicationName>toSonatypeRepository` should get the credentials for the `sonatypeRepository`. 
If we were to add the mavenPublication in the `external-publish-gradle-plugin` similar to:
https://github.com/palantir/gradle-external-publish-plugin/blob/develop/src/main/java/com/palantir/gradle/externalpublish/ExternalPublishJarPlugin.java#L33-L35
it works. I guess this is the part we are missing in the plugin: https://github.com/gradle-nexus/publish-plugin?tab=readme-ov-file#add-metadata.

Reply:
Hmm, I'm a bit confused by all this. I think the Gradle plugin publish plugin is setting up a publication for us (`pluginMaven`?). Publishing to the Gradle plugin portal shouldn't require anything to do to with sonatype though - that's just for publishing to maven central. I guess this only need to hold up this PR if we now need to add the jar... I'll take a look and investigate.

Reply:
Ok, I think I worked out what's going wrong. The Gradle plugin did not apply the `ExternalPublishBasePlugin`, which has a method `disableOtherPublicationsFromPublishingToSonatype` that stops non-sonatype publications from trying to publish to sonatype. The Gradle plugin publish plugin's publication was trying to be published to sonatype, but didn't have the creds as it never got them from the sonatype plugin. Now it is appropriately ignored.